apps/postgres-new/components/ExpandableTextArea.tsx

```diff
@@ -0,0 +1,51 @@
+'use client'
+
+import React, { forwardRef, useEffect, useImperativeHandle, useRef } from 'react'
+import { cn } from '~/lib/utils'
+import { TextArea } from './ui/textarea'
+
+export interface ExpandingTextAreaProps extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
+  /* The value of the textarea. Required to calculate the height of the textarea. */
+  value: string
+}
+
+/**
+ * This is a custom TextArea component that expands based on the content.
+ */
+const ExpandingTextArea = forwardRef<HTMLTextAreaElement, ExpandingTextAreaProps>(
+  ({ className, value, ...props }, ref) => {
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+    // Expose the ref to the parent component
+    useImperativeHandle(ref, () => textAreaRef.current!)
```

Comment:
Any reason we need to wrap this in an imperative handle vs using `ref` directly on the `textarea`?